File: frontend/src/types/work-item-template.ts
export interface WorkItemTemplate {
  id: number
  name: string
  description?: string
  issue_type_id?: number
  issue_type?: {
    id: number
    name: string
    color: string
    icon: string
  }
  defaults: Record<string, any>
  is_default: boolean
  project_id: number
  workspace_id: number
  created_at: string
  updated_at: string
}

export interface WorkItemTemplateCreate {
  name: string
  description?: string
  issue_type_id?: number
  defaults?: Record<string, any>
  is_default?: boolean
}

export interface WorkItemTemplateUpdate {
  name?: string
  description?: string
  issue_type_id?: number
  defaults?: Record<string, any>
  is_default?: boolean
}

File: backend/internal/dto/response/work_item_template.go
package response

import (
	"encoding/json"
	"time"
)

type WorkItemTemplateResponse struct {
	ID          uint64          `json:"id"`
	Name        string          `json:"name"`
	Description string          `json:"description"`
	IssueTypeID *uint64         `json:"issue_type_id"`
	IssueType   *IssueTypeLite  `json:"issue_type,omitempty"`
	Defaults    json.RawMessage `json:"defaults"`
	IsDefault   bool            `json:"is_default"`
	ProjectID   uint64          `json:"project_id"`
	WorkspaceID uint64          `json:"workspace_id"`
	CreatedAt   time.Time       `json:"created_at"`
	UpdatedAt   time.Time       `json:"updated_at"`
}
